aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -165,6 +191,16 @@ impl GasAlgebra for StandardGasAlgebra {
         &mut self,
         abstract_amount: impl GasExpression<VMGasParameters, Unit = InternalGasUnit> + Debug,
     ) -> PartialVMResult<()> {
+        self.counter_for_kill_switch += 1;
```

Comment:
This is quite ugly but can work for now, I wonder if removing `?Sized` will help with perf here

Reply:
Sized was being lost along the line inside some templated FnOnce make gas meter, found it and fixed. should be static dispatch now.